SecurityPkg/Library/Tpm2DeviceLibDTpm/Tpm2Svsm.c

```diff
@@ -0,0 +1,128 @@
+/** @file
+  SVSM TPM communication
+
+Copyright (C) 2024 James.Bottomley@HansenPartnership.com
+Copyright (C) 2024 IBM Corporation
+
+SPDX-License-Identifier: BSD-2-Clause-Patent
+
+**/
+
+#include <Library/BaseLib.h>
+#include <Library/BaseMemoryLib.h>
+#include <Library/DebugLib.h>
+#include <Library/Tpm2DeviceLib.h>
+#include <Library/PcdLib.h>
+#include <Library/AmdSvsmLib.h>
+
+#include "Tpm2DeviceLibDTpm.h"
+
+/**
+  Platform commands (MSSIM commands) can be sent through the
+  SVSM_VTPM_CMD operation. Each command can have its own
+  request and response structures.
+**/
+#define TPM_SEND_COMMAND  8
+
+/* Max req/resp buffer size */
+#define TPM_PLATFORM_MAX_BUFFER  4096
+
+STATIC UINT8  Tpm2SvsmBuffer[TPM_PLATFORM_MAX_BUFFER];

```

Comment:
Are commands sent prior to having access to dynamic memory allocation? This expands the flash description size another 4KiB and kicks over some limits we have.

Reply:
Is that a new issue? Does it work with one of the previous versions of these patches?

Reply:
Well, if we don't have to allocate a 4k buffer statically we should not do that, but go `AllocatePool` + `FreePool` instead.

One common trick which can be used to allocate memory in case the pool allocator is not up yet is to create a HOB instead.  Drawback is that HOBs can't be freed, but it should be possible to have a small communication buffer to probe for the svsm vtpm protocol being present and only create a (larger) HOB in case we actually need it.